packages/taiko-client/driver/preconf_blocks/api.go

```diff
@@ -275,31 +277,29 @@ func (s *PreconfBlockAPIServer) BuildPreconfBlock(c echo.Context) error {
 				"baseFee", header.BaseFee,
 			)
 		} else {
-			// sign the block hash, persist it to L1Origin as the signature
-			sigBytes, err := s.p2pSigner.Sign(
-				ctx,
-				p2p.SigningDomainBlocksV1,
-				s.rpc.L2.ChainID,
-				header.Hash().Bytes(),
-			)
-			if err != nil {
-				log.Warn(
-					"Failed to sign the preconfirmation block payload",
-					"blockHash", executablePayload.BlockHash.Hex(),
-					"blockID", header.Number.Uint64(),
-				)
-				return s.returnError(c, http.StatusInternalServerError, fmt.Errorf("failed to sign payload: %w", err))
-			}
-
-			if _, err = s.rpc.L2Engine.SetL1OriginSignature(ctx, header.Number, *sigBytes); err != nil {
-				return s.returnError(
-					c,
-					http.StatusInternalServerError,
-					fmt.Errorf("failed to update L1 origin signature: %w", err),
+			var sigBytes *[65]byte
+			// For pre-Shasta, we still persist the signature to L1Origin.
+			if header.Number.Uint64() < s.rpc.ShastaClients.ForkHeight.Uint64() && s.p2pSigner != nil {
+				b, err := s.p2pSigner.Sign(
+					ctx,
+					p2p.SigningDomainBlocksV1,
+					s.rpc.L2.ChainID,
```

Comment:
yeah lets keep the original API unchanged here and just add a new post-shasta post-whitelist API endpoint